test/integration/hermetic_test.go

```diff
@@ -76,6 +86,7 @@ func TestKubeInferenceModelRequest(t *testing.T) {
 		wantHeaders       []*configPb.HeaderValueOption
 		wantMetadata      *structpb.Struct
 		wantBody          []byte
+		wantMetrics       string
```

Comment:
The wantMetrics relies on 1) the test cases are executed sequentially (i.e., do not use t.Parallel()) and 2. the order of the test cases.

I think this is very inconvenient. For example, if someone adds a new test cases at the beginning, then all following wantMetrics need to be updated. 

One option is to start a new server every test, but this can be slow. Alternatively, is it possible to clear the metrics after a test case is done?

Reply:
Done. Reset the metrics every run.